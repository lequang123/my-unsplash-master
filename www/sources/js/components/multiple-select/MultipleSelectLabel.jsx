import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { MAX_SELECTED_ITEM_FOR_DISPLAY } from './constants';


export default class MultipleSelectLabel extends Component {
    static propTypes = {
        textOptions: PropTypes.object,
        selectedItems: PropTypes.arrayOf(PropTypes.object).isRequired,
        maxDisplayItemCount: PropTypes.number
    };

    static defaultProps = {
        maxDisplayItemCount: MAX_SELECTED_ITEM_FOR_DISPLAY
    };

    constructor(props) {
        super(props);
    }

    get selectedItemsString() {
        const { selectedItems, maxDisplayItemCount, textOptions } = this.props;
        const selectedValues = selectedItems.map(item => item.value);

        let displayText = textOptions['SelectOptions'];
        const selectedItemCount = selectedValues.length;

        if (this.props.isAllTextShown && selectedItemCount === this.props.dataSourceSize) {
            displayText = textOptions['Texts'];
        } else if (selectedItemCount > maxDisplayItemCount) {
            displayText = textOptions['SelectedItemCount'].replace('#', selectedItemCount);
        } else if (selectedItemCount >= 1) {
            displayText = selectedValues.join(', ');
        } else if(!this.props.dataSourceSize){
            displayText = textOptions['NoData'];
        }

        return displayText;
    }

    render() {
        return (
            <button type='button' disabled = {this.props.isDisable} className='multiple-select-default multiple-select-label'
                onClick={this.props.onToggle}
            >
                <span className='text-display'>{this.selectedItemsString}</span>
                <b className='caret' />
            </button>
        );
    }
}