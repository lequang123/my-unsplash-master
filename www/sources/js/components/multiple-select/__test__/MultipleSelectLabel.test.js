import React from 'react';
import MultipleSelectLabel from '../MultipleSelectLabel';
import { shallow } from 'enzyme';

describe('MultipleSelectLabel', () => {
    it('should have correct snapshots', () => {
        // Arrange
        const props = {
            selectedItems: [],
            textOptions: {}
        };

        // Action
        const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);

        // Assert
        expect(multipleSelectLabel).toMatchSnapshot();
    });

    describe('selectedItemsString', () => {
        it('should return All when isAllTextShown true and dataSourceSize equal to selectedItems length', () => {
            // Arrange
            const props = {
                isAllTextShown: true,
                dataSourceSize: 0,
                selectedItems: [],
                textOptions: {
                    Texts: 'Texts'
                }
            }
    
            const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);
    
            // Action
            const result = multipleSelectLabel.instance().selectedItemsString;

            // Assert
            expect(result).toEqual(props.textOptions.Texts);
        });
        
        it('should return SelectedItemCount when selectedItemCount greater than maxDisplayItemCount', () => {
            // Arrange
            const props = {
                maxDisplayItemCount: 0,
                selectedItems: [
                    {
                        value: 'value'
                    }
                ],
                textOptions: {
                    SelectedItemCount: '# items'
                }
            }
    
            const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);
    
            // Action
            const result = multipleSelectLabel.instance().selectedItemsString;

            // Assert
            expect(result).toEqual('1 items');
        });
        
        it('should return selected value when selectedItemCount not less than 1', () => {
            // Arrange
            const props = {
                maxDisplayItemCount: 10,
                selectedItems: [
                    {
                        value: 'value'
                    },
                    {
                        value: 'value1'
                    }
                ],
                textOptions: {}
            }
    
            const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);
    
            // Action
            const result = multipleSelectLabel.instance().selectedItemsString;

            // Assert
            expect(result).toEqual('value, value1');
        });
        
        it('should return no data value when dataSourceSize equal 0', () => {
            // Arrange
            const props = {
                maxDisplayItemCount: 10,
                selectedItems: [],
                textOptions: {
                    NoData: 'NoData'
                },
                dataSourceSize: 0
            }
    
            const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);
    
            // Action
            const result = multipleSelectLabel.instance().selectedItemsString;

            // Assert
            expect(result).toEqual(props.textOptions.NoData);
        });
        
        it('should return SelectOptions when not having selectedItem', () => {
            // Arrange
            const props = {
                selectedItems: [],
                textOptions: {
                    SelectOptions: 'SelectOptions'
                },
                dataSourceSize: 10
            }
    
            const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);
    
            // Action
            const result = multipleSelectLabel.instance().selectedItemsString;

            // Assert
            expect(result).toEqual(props.textOptions.SelectOptions);
        });
    });

    it('render should return correct components', () => {
        // Arrange
        const props = {
            selectedItems: [],
            onToggle: jest.fn(),
            textOptions: {
                SelectOptions: 'SelectOptions'
            },
            dataSourceSize: 10
        }

        // Action
        const multipleSelectLabel = shallow(<MultipleSelectLabel {...props} />);

        // Assert
        expect(multipleSelectLabel.find('button').props().type).toBe('button');
        expect(multipleSelectLabel.find('button').props().className).toBe('multiple-select-default multiple-select-label');
        expect(multipleSelectLabel.find('button').props().onClick).toEqual(props.onToggle);
        expect(multipleSelectLabel.find('span').props().className).toBe('text-display');
        expect(multipleSelectLabel.find('span').html()).toContain(props.textOptions.SelectOptions);
        expect(multipleSelectLabel.find('b').html()).toBe('<b class="caret"></b>');
    });
});
