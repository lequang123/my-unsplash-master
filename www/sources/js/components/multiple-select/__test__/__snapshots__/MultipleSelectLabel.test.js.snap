// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultipleSelectLabel should have correct snapshots 1`] = `ShallowWrapper {}`;
